### PR TITLE
std: Add std.math.divCeil

### PR DESCRIPTION
**Pull request by katesuyu**
This pull request adds an implementation of `divCeil` to the standard library.

Ceiled division is a useful operation that is easy to implement in a naive manner, but history has shown that not having it in a language's standard library leads to subpar attempts to implement the primitive. For example, both implementations [here](https://stackoverflow.com/a/2745086) not only ignore the existence of negative numbers, but also have overflow bugs. The first implementation addresses the overflow by referring to the second, which suggests checking for equality with 0 to avoid overflowing the numerator. But that approach *still* completely ignores negative numbers, and a naive implementation might incorrectly return `0` for `0/0` rather than `error.DivisionByZero`!

This standard implementation gives one obvious way for users in need of ceiled division to perform the required operation without the undesired behavior that could occur in a homemade implementation.

### CODE REVIEW

File: lib/std/math.zig [hunk 1]
```diff
@@ -621,6 +621,30 @@ fn testDivFloor() void {
...
+    }
+    return (try divFloor(T, numerator - 1, denominator)) + 1;
+}
+
+test "math.divCeil" {
+    testDivCeil();
+    comptime testDivCeil();
+}
+fn testDivCeil() void {
+    testing.expect((divCeil(i32, 5, 3) catch unreachable) == 2);
```
Comment (ranma42) [inline lib/std/math.zig:639]: I would recommend testing all of the sign combinations
```suggestion
    testing.expect((divCeil(i32, -5, -3) catch unreachable) == 2);
    testing.expect((divCeil(i32, 5, -3) catch unreachable) == 2);
```

Comment (katesuyu) [reply lib/std/math.zig:639]: Fair point... drat I really thought it'd be this simple.

**Comment by daurnimator** [comment]
Consider adding tests with `comptime_float` and `comptime_int` as well.

You may want
```zig
switch(@typeInfo(T)) {
  .Int, .ComptimeInt => |int_info| ......,
  .Float, .ComptimeFloat => |float_info| ......,
  else => @compileError("divCeil unsupported on " ++ @typeName(T)),
}
```

**Comment by katesuyu** [comment]
I've added those tests and reorganized the code to have a bit clearer control flow. Previously I assumed that if the type was not a float, it would be an int. That ignored comptime floats, though, plus it makes for cleaner code to use switches when there are multiple possible type tags.

File: lib/std/math.zig [hunk 2]
```diff
@@ -621,6 +621,59 @@ fn testDivFloor() void {
...
+    testing.expect((divCeil(i32, -5, 3) catch unreachable) == -1);
+    testing.expect((divCeil(i32, 5, -3) catch unreachable) == -1);
```
Comment (daurnimator) [inline lib/std/math.zig:651]: prefer `testing.expectEqual` for this sort of test (other code in this file is quite old and pre-dates the introduction of `expectEqual`)

**Review by daurnimator — APPROVED** [review]

File: lib/std/math.zig (hunk 2 above)
Comment (katesuyu) [reply lib/std/math.zig:651]: Sure thing! I was just afraid to be the only `div` function that used that :worried:

Comment (daurnimator) [reply lib/std/math.zig:651]: I don't think anyone would complain if you felt like updating the whole file ;) (in a separate commit of course)

Comment (katesuyu) [reply lib/std/math.zig:651]: I'll just update my test for now.

File: lib/std/math.zig [hunk 3]
```diff
@@ -646,31 +646,31 @@ test "math.divCeil" {
     comptime testDivCeil();
 }
 fn testDivCeil() void {
-    testing.expect((divCeil(i32, 5, 3) catch unreachable) == 2);
-    testing.expect((divCeil(i32, -5, 3) catch unreachable) == -1);
-    testing.expect((divCeil(i32, 5, -3) catch unreachable) == -1);
-    testing.expect((divCeil(i32, -5, -3) catch unreachable) == 2);
-    testing.expect((divCeil(i32, 0, 5) catch unreachable) == 0);
-    testing.expect((divCeil(u32, 0, 5) catch unreachable) == 0);
+    testing.expectEqual(divCeil(i32, 5, 3) catch unreachable, 2);
```
Comment (daurnimator) [inline lib/std/math.zig:649]: `expectEqual` takes first argument as the expected value (with correct type) and then the value to be tested.

**Pull request merged**